l10n_ch_sepa/models/account_payment_order.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# © 2016 Akretion (www.akretion.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, _
+from openerp.exceptions import UserError
+from lxml import etree
+
+
+class AccountPaymentOrder(models.Model):
+    _inherit = 'account.payment.order'
+
+    @api.multi
+    def compute_sepa_final_hook(self, sepa):
+        self.ensure_one()
+        sepa = super(AccountPaymentOrder, self).compute_sepa_final_hook(sepa)
+        # BVR orders cannot be SEPA orders
+        if sepa:
+            for line in self.payment_line_ids:
+                if line.communication_type == 'bvr':
+                    sepa = False
+                    break
```

Comment:
What about

          sepa &= not any(line.communication_type == 'bvr'
                          for line in self.payment_line_ids)